src/lib/storage/table.cpp

```diff
@@ -19,74 +19,58 @@
 namespace opossum {
 
 Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  _target_chunk_size = target_chunk_size;
+  create_new_chunk();
 }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+  // add column to existing last chunk
+  // TODO(all): add column to all existing chunks?
+  _chunks.back()->create_and_add_segment(type);
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (_chunks.back()->size() >= target_chunk_size()) {
+    create_new_chunk();
+  }
+  _chunks.back()->append(values);
 }
 
 void Table::create_new_chunk() {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto new_chunk = std::make_shared<Chunk>();
```

Comment:
You could use `const`, it should mark the pointer and not the chunk object itself.